stringbuilder.go

```diff
@@ -289,6 +289,19 @@ func (s *StringBuilder) Reverse() *StringBuilder {
 	return s
 }
 
+// Substring creates a substring from the string builder using a start and an end(exclusive) bound
+func (s *StringBuilder) Substring(start, end int) (string, error) {
+	if start < 0 {
+		return "", fmt.Errorf("start should always be greater than or equal to zero")
+	}
+	if end > s.position {
+		return "", fmt.Errorf("end cannot be greater than the length of string builder")
+	}
```

Comment:
There should also be a check if `start < end`.

Reply:
Ahh nice catch. Updated with additional check and a test